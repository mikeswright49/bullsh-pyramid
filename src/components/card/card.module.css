.card {
    height: 123px;
    width: 79px;
    border-radius: 6px;
    background-image: url('/card-deck.png');
    background-repeat: no-repeat;
    background-position: 0px 0px;
}

/*
* Yes important.
* We're setting the position for the cards based on data in a style object
**/
.cardHidden {
    background-position: -158px -492px !important;
}
