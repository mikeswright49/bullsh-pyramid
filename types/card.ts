export type Card = {
    value: number;
    suit: number;
    hidden: boolean;
};
