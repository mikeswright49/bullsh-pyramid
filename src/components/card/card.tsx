import React from 'react';
import styles from './card.module.css';
import cn from 'classnames';
import { Card as CardType } from 'types/card';

const CARD_HEIGHT = 123;
const CARD_WIDTH = 79;

export function Card({ card }: { card: CardType }): JSX.Element {
    return (
        <div
            style={{
                backgroundPosition: `-${card.value * CARD_WIDTH}px -${card.suit * CARD_HEIGHT}px`,
            }}
            className={cn('flex-center-center', styles.card, { [styles.cardHidden]: card.hidden })}
        ></div>
    );
}
