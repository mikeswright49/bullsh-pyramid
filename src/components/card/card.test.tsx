import { render } from '@testing-library/react';
import React from 'react';
import { Card } from './card';

describe('<Unit Test> Card', () => {
    it('should be able to render the component', () => {
        const { container } = render(<Card card={{ value: 1, suit: 2, hidden: false }} />);
        expect(container).toMatchSnapshot();
    });
    it('shold be able to change styles on visibility', () => {
        const { container } = render(<Card card={{ value: 1, suit: 2, hidden: true }} />);
        expect(container).toMatchSnapshot();
    });
});
