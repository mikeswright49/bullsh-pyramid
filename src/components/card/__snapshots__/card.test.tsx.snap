// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Unit Test> Card shold be able to change styles on visibility 1`] = `
<div>
  <div
    class="flex-center-center card cardHidden"
    data-testid="card-2-1"
    style="background-position: -79px -246px;"
  />
</div>
`;

exports[`<Unit Test> Card should be able to render the component 1`] = `
<div>
  <div
    class="flex-center-center card"
    data-testid="card-2-1"
    style="background-position: -79px -246px;"
  />
</div>
`;
